src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -65,6 +65,22 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                     {
                         ConfigureTempApiHashKey(builder, settings.Authentication);
                     }
+
+                    // User-specified configuration file path is considered highest precedence, but does NOT override other configuration sources
+                    string userFilePath = settings.UserProvidedConfigFilePath;
+
+                    if (File.Exists(userFilePath) && Path.GetExtension(userFilePath).ToLower().Equals(".json"))
+                    {
+                        try
+                        {
+                            File.OpenRead(userFilePath).Dispose(); // If this succeeds, we have read-permissions
+                            builder.AddJsonFile(userFilePath, optional: true, reloadOnChange: true);
+                        }
+                        catch (Exception)
```

Comment:
This seems like a reasonable error to log. Is there a reason we wouldn't want to do this?